File: src/main/java/ar/edu/unq/epers/unidad4/model/Item.java
package ar.edu.unq.epers.unidad4.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;


@Data
@EqualsAndHashCode(exclude = "owner")
@ToString(exclude = "owner")
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nombre;
    private int peso;

    @ManyToOne
    private Personaje owner;

    public Item(String nombre, int peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/model/Item.kt
package ar.edu.unq.epers.unidad4.model

import org.springframework.data.neo4j.core.schema.GeneratedValue
import org.springframework.data.neo4j.core.schema.Id
import org.springframework.data.neo4j.core.schema.Node
import org.springframework.data.neo4j.core.schema.Relationship

@Node
class Item() {

    @Id
    @GeneratedValue
    var id: Long? = null
    var nombre: String = ""
    var peso: Int = 0

    @Relationship(type = "POSEE", direction = Relationship.Direction.INCOMING)
    var owner: Personaje? = null

    constructor(nombre: String, peso: Int) : this() {
        this.nombre = nombre
        this.peso = peso
    }

}